File: 2024/AdventOfCode/Tests/Day05Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day05Test {
    private const string Example = "Inputs/05-Example.txt";
    private const string Input = "Inputs/05.txt";
    private static Day05 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day05(Example);
            const string expected = "143";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day05(Input);
            const string expected = "4281";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day05(Example);
            const string expected = "123";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day05(Input);
            const string expected = "5466";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}

File: 2025/test/advent_of_code/day_05_test.exs
defmodule AdventOfCode.Day05Test do
  use ExUnit.Case

  import AdventOfCode.Day05

  @day 5
  @year 2025

  @example String.trim("""
           3-5
           10-14
           16-20
           12-18

           1
           5
           8
           11
           17
           32
           """)

  test "part1 example" do
    result = part1(@example)

    assert result == 3
  end

  test "part1" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part1(input)

    assert result == 652
  end

  test "part2 example" do
    result = part2(@example)

    assert result == 14
  end

  test "part2" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part2(input)

    assert result == 341_753_674_214_273
  end
end
